src/tty/mod.rs

```diff
@@ -39,6 +40,7 @@ pub trait Renderer {
         old_layout: &Layout,
         new_layout: &Layout,
         highlighter: Option<&dyn Highlighter>,
+        vi_mode: Option<InputMode>,
```

Comment:
You can factorize code by passing a `PromptState` instead of `Option<InputMode>`.